test/unit/core.js

```diff
@@ -1217,6 +1217,25 @@ test("jQuery.each(Object,Function)", function() {
 	equal( i, 2, "Iteration over document.styleSheets" );
 });
 
+test( "JIT compilation does not interfere with length retrieval (gh-2145)", function() {
+	expect( 4 );
+
+	var i;
+
+	// Trigger JIT
```

Comment:
"try to convince JSC to use one of its optimizing compilers by providing code which can be LICM'd into nothing", I think?